algebird-core/src/main/scala/com/twitter/algebird/Monoid.scala

```diff
@@ -71,7 +71,7 @@ class OptionMonoid[T](implicit semi: Semigroup[T]) extends Monoid[Option[T]] {
   }
   override def sumOption(items: TraversableOnce[Option[T]]): Option[Option[T]] =
     if (items.isEmpty) None
-    else Some(semi.sumOption(items.filter(_.isDefined).map { _.get }))
+    else Some(semi.sumOption(items.flatten))
```

Comment:
there are several choices here. For algebird internals, we really try to have benchmarks (see the benchmark subproject) to make the choice. Some alternatives:
```scala
// we can try with or without .toIterator, which may force some laziness we might want
items.toIterator.flatMap(identity)
items.toIterator.collect { case Some(t) => t }
```